clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -356,4 +361,15 @@ public static void assertFutureError(Future<?> future, Class<? extends Throwable
                 exceptionClass, cause.getClass());
         }
     }
+
+    public static boolean maybeBeginServerReauthentication(KafkaChannel channel, NetworkReceive networkReceive, Time time) {
```

Comment:
👍 True that it is not really reusable; `NioEchoServer` needs that functionality, so now it's a `private static boolean` method on that class rather than a `public static boolean` method on `TestUtils`.

Reply:
The above change resulted in a disallowed import error, so I refactored out the following method and put it back into `TestUtils`. This is better than having the whole `maybeBeginServerReauthentication()` method there.
```
    public static ApiKeys apiKeyFrom(NetworkReceive networkReceive) {
        return RequestHeader.parse(networkReceive.payload().duplicate()).apiKey();
    }
```